cmd/openshift-install/migrate/azure/migrate.go

```diff
@@ -0,0 +1,47 @@
+package azure
+
+import (
+	"github.com/spf13/cobra"
+
+	azmigrate "github.com/openshift/installer/pkg/migrate/azure"
+)
+
+var (
+	azureMigrateOpts struct {
+		zone              string
+		oldResourceGroup  string
+		newResourceGroup  string
+		virtualNetwork    string
+		vnetResourceGroup string
+		link              bool
+		public            bool
+	}
+)
+
+func runMigrateAzurePrivateDNSMigrateCmd(cmd *cobra.Command, args []string) error {
+	if (azureMigrateOpts.zone == "" || azureMigrateOpts.oldResourceGroup == "") || (azureMigrateOpts.link == true && azureMigrateOpts.virtualNetwork == "") || (azureMigrateOpts.virtualNetwork != "" && azureMigrateOpts.vnetResourceGroup == "") {
+		return cmd.Help()
+	}
+
+	return azmigrate.Migrate(azureMigrateOpts.oldResourceGroup, azureMigrateOpts.newResourceGroup, azureMigrateOpts.zone, azureMigrateOpts.virtualNetwork, azureMigrateOpts.vnetResourceGroup, azureMigrateOpts.link, azureMigrateOpts.public)
+}
+
+// NewMigrateAzurePrivateDNSMigrateCmd adds the migrate command to openshift-install
+func NewMigrateAzurePrivateDNSMigrateCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "migrate",
+		Short: "Migrate a zone",
+		Long:  "This will migrate a legacy private zone to a new style private zone.",
+		RunE:  runMigrateAzurePrivateDNSMigrateCmd,
+	}
+
+	cmd.PersistentFlags().StringVar(&azureMigrateOpts.zone, "zone", "", "The zone to migrate")
+	cmd.PersistentFlags().StringVar(&azureMigrateOpts.oldResourceGroup, "oldResourceGroup", "", "The resource group of the legacy zone")
```

Comment:
I think the fails shouldn't be in camel case, rather `-` separated

Reply:
Updated.